src/scripts/selectors/PlayerSelectors.js

```diff
@@ -33,26 +33,30 @@ export const getPlayingVideo: (
   }
 )
 
+export const getFormattedDuration: (
+  state: RootState
+) => string = createSelector(
+  [getPlayingVideo],
+  (playingVideo: ?VideoRecord): string => {
+    const duration: ?string = playingVideo && playingVideo.get('duration')
+    return formatDuration(duration)
+  }
+)
+
 export const getFormattedCurrentTime: (
   state: RootState
 ) => string = createSelector(
-  [getPlayerCurrentTimeSeconds],
-  (currentTimeSeconds: number): string => {
+  [getPlayerCurrentTimeSeconds, getFormattedDuration],
+  (currentTimeSeconds: number, formattedDuration: string): string => {
     const roundedTime: number = Math.floor(currentTimeSeconds)
-    return TimeFormat.fromS(
+    const formattedCurrentTime: string = TimeFormat.fromS(
       roundedTime,
       roundedTime >= 3600 ? 'hh:mm:ss' : 'mm:ss'
     )
-  }
-)
 
-export const getFormattedDuration: (
-  state: RootState
-) => string = createSelector(
-  [getPlayingVideo],
-  (playingVideo: ?VideoRecord): string => {
-    const duration: ?string = playingVideo && playingVideo.get('duration')
-    return formatDuration(duration)
+    return formattedDuration < formattedCurrentTime
+      ? formattedDuration
+      : formattedCurrentTime
```

Comment:
Return the duration if duration is less than what we have calculated as the current time